lib/cloudant.js

```diff
@@ -901,43 +901,116 @@ Cloudant.prototype.selectModel = function(model, migrate) {
+ * `automigrate` does that.
  * @param {String[]} [models] A model name or an array of model names. If not
  * present, apply to all models
  * @param {Function} [cb] The callback function
  */
 Cloudant.prototype.autoupdate = function(models, cb) {
   debug('Cloudant.prototype.autoupdate %j', models);
-  this.automigrate(models, cb);
+  var self = this;
+  async.eachSeries(models, function(model, cb2) {
+    debug('Cloudant.prototype.autoupdate model %j', model);
+    var mo = self.selectModel(model, true);
+    self.updateIndex(mo, model, cb2);
+  }, function(err) {
+    debug('Cloudant.prototype.autoupdate %j', err);
+    cb(err);
+  });
 };
 
 /**
  * Perform automigrate for the given models.
  * Notes: Cloudant stores a Model as a design doc in database,
  * `ddoc` is the doc's _id, auto-generated as 'lb-index-ddoc' + modelName,
  * `name` is the index name, auto-generated as 'lb-index' + modelName.
- * It does NOT store properties in the doc, which makes autoupdate actually
- * does the same thing as automigrate.
- * It does NOT destroy previous model instances if model exists, user needs
- * to manually destroy them if necessary.
- *
- * @param {String|String[]} [models] A model name or an array of model names. If not
- * present, apply to all models
+ * It does NOT store properties in the doc.
+ * `automigrate` destroys model data if the model exists.
+ * @param {String|String[]} [models] A model name or an array of model names.
+ * If not present, apply to all models
  * @param {Function} [cb] The callback function
  */
 Cloudant.prototype.automigrate = function(models, cb) {
   debug('Cloudant.prototype.automigrate models %j', models);
   var self = this;
-  async.eachSeries(models, function(model, cb2) {
-    debug('Cloudant.prototype.automigrate model %j', model);
-    var mo = self.selectModel(model, true);
-    self.updateIndex(mo, model, cb2);
+  var existingModels = [];
+  async.series([
+    function(callback) {
+      checkExistingModels(callback);
+    },
+    function(callback) {
+      destroyData(callback);
+    },
+    function(callback) {
+      self.autoupdate(models, callback);
+    }], cb);
+  function checkExistingModels(checkCb) {
+    async.eachSeries(models, function(model, cb) {
+      self.isActual(model, function(err, exist) {
+        if (err) return cb(err);
+        if (exist) existingModels.push(model);
+        cb();
+      });
+    }, function(err) {
+      debug('Cloudant.prototype.automigrate checkExistingModels %j', err);
+      checkCb(err);
+    });
+  };
+  function destroyData(destroyCb) {
+    async.eachSeries(existingModels, function(model, cb) {
+      self.destroyAll(model, {}, {}, cb);
+    }, function(err) {
+      debug('Cloudant.prototype.automigrate %j', err);
+      destroyCb(err);
+    });
+  };
+};
+
+/**
+  * Check if the models exist
+  * @param {String|String[]} [models] A model name or an array of model names. * If not present, apply to all models
```

Comment:
Was the second part of this sentence suppose to be on a new line? I see an asterisk.